R/rd-html.R

```diff
@@ -583,3 +584,17 @@ is_newline <- function(x, trim = FALSE) {
   }
   identical(text, "\n")
 }
+
+fix_R6_inherited_hrefs <- function(x) {
```

Comment:
Could this use xml2 rather than regular expressions? It seems like the HTML should contain all the data we need, e.g. `<span class="pkg-link" data-pkg="R6test" data-topic="Animal" data-id="initialize">`

Reply:
So, the data here are of the form:
```r
c(
  "",
  " ",
  "<span class=\"pkg-link\" data-pkg=\"R6test\" data-topic=\"Animal\" data-id=\"initialize\">",
  "<a href='../../R6test/html/Animal.html#method-initialize'><code>R6test::Animal$initialize()</code></a>",
  "</span>"
)
```
I agree that using `xml2` to parse HTML is in general a better idea than regular expressions, but I tried multiple things and didn't really find a good approach here. Joining the character vector into one string just so it can be read by `xml2::read_html`, then fixing the urls, and afterward splitting the string back into chunks is possible, but doesn't seem optimal. If you know a better approach please let me know. If there is a better place in the code where the reference is already read as xml, then something like `xml2::xml_find_all("//*[@class=\"pkg-link\"]//*[@href]")` should return all the nodes with package links that need modification though.

Reply:
Hmmmm, I see. In that case, I'd prefer to leave this sit for a while so I can contemplate a cleaner solution (if one exists)

Reply:
So I had another look at it and I think an `xml2` based solution should just be implemented in a completely different spot (see last commits), namely in `render_page`. At that point, the code is doing multiple times:
```r
if (condition) {
  html <- xml2::read_html(rendered)
  # some function that applies fixes to the html with xml2
  rendered <- as.character(html, options = character())
}
```
The nice thing about fixing the links here is that with the condition `pkg$desc$has_dep("R6")` people only pay the price for fixing these links if they actually have R6 as a dependency. If more instances appear where these kind of links need to be fixed then you only need to adjust the if statement. The code now also no longer uses regular expressions.